### PR TITLE
chore!: use AWS CDK 2

### PR DESCRIPTION
**Pull request by completenovice**
This updates the CDK package to version 2. This is to take advantage
of new features not available in CDK 1.

Co-authored-by: nchylak <nadia.chylak@avrios.com>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,8 +20,8 @@
     "typescript": "4.2.3"
   },
   "peerDependencies": {
-    "@aws-cdk/aws-cloudformation": "^1.91.0",
-    "@aws-cdk/cloudformation-include": "^1.91.0",
-    "@aws-cdk/core": "^1.91.0"
+    "@aws-cdk/assert": "2.8.0",
```
Comment (jaeggir) [inline package.json:23]: they already published version `2.12.0` 😅

**Review by jaeggir — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -9,7 +9,9 @@
     "test": "jest"
   },
   "devDependencies": {
-    "@aws-cdk/assert": "^1.91.0",
+    "@aws-cdk/assert": "2.12.0",
```
Comment (jaeggir) [inline package.json:12]: ah just saw this. I think you need only `assert` as a `devDependency`. cdk-lib and construct are only needed in `peerDependencies` (where assert is not needed :D). But not 100% sure, so maybe give it a quick try.

Comment (completenovice) [reply package.json:12]: You're right! It does work. Good spot 👌

File: package.json [hunk 3]
```diff
@@ -18,8 +18,7 @@
     "typescript": "4.2.3"
   },
   "peerDependencies": {
-    "@aws-cdk/aws-cloudformation": "^1.91.0",
-    "@aws-cdk/cloudformation-include": "^1.91.0",
-    "@aws-cdk/core": "^1.91.0"
+    "aws-cdk-lib": "2.12.0",
+    "constructs": "10.0.63"
```
Comment (completenovice) [inline package.json:22]: Would this be a place to use `^` - seeing as downstream consumers may be on a version later than 2.12.0 or 10.0.63?

File: .nvmrc [hunk 4]
```diff
@@ -0,0 +1 @@
+15
```
Comment (jaeggir) [inline .nvmrc:1]: LTS version is `16`. I think we should only use `14` (legacy) and `16` (for everything new). This project already uses lockfile version 2, so it should use node 16 (-> npm 8)

Comment (completenovice) [reply .nvmrc]: Cool, I wasn't sure which one to use so went with the same as cdk-utils. I'll make sure we use 16 going forward, where lockfile version is 2 👍

**Pull request merged**